data_structures/stacks/postfix_evaluation.py

```diff
@@ -17,52 +24,211 @@
         Result =  59
 """
 
-import operator as op
+# Defining valid unary operator symbols
+UNARY_OP_SYMBOLS = ("-", "+")
+
+# Defining valid binary operator symbols
+BINARY_OP_SYMBOLS = ("-", "+", "*", "^", "/")
+
+
+def get_number(data: str) -> int | float | str:
+    """
+    Converts the given data to appropriate number if it is indeed a number, else returns
+    the data as it is with a False flag. This function also serves as a check of whether
+    the input is a number or not.
+
+    Parameters
+    ----------
+    data : str
+        The data which needs to be converted to the appropriate number
+
+    Returns
+    -------
+    bool,  int or float
+        Returns a tuple of (a, b) where 'a' is True if data is indeed a number (integer
+        or numeric) and 'b' is either an integer of a floating point number.
+        If 'a' is False, then b is 'data'
+    """
+    try:
+        return int(data)
+    except ValueError:
+        try:
+            return float(data)
+        except ValueError:
+            if is_operator(data):
+                return data
+            msg = f"{data} is neither a number nor a valid operator"
+            raise ValueError(msg)
+
+
+def is_operator(data: str | int | float) -> bool:
```

Comment:
Hi @tianyizheng02, `is_operator` can have float inputs apart from strings as it takes each token from the `valid expression` list. The `valid_expression` list is made using the `parse_token` function which returns strings if it's an operator, or a float if it's a number as shown below-
![image](https://github.com/TheAlgorithms/Python/assets/12524022/a7a13001-655b-4d6a-afdc-c7c73e5fc829)
.
Hence it is necessary to give the float type along with str.

Reply:
I see, that makes sense. In that case, the function header would be
```suggestion
def is_operator(token: str | float) -> bool:
```
1) No `int` because I believe ruff now enforces `int | float` -> `float`
2) Let's rename the variable to `token` to make it clearer that the input is a token (which is allowed to be a number)